File: extract_pdf.mjs
import fs from 'fs';
import PDFParser from 'pdf2json';

const pdfParser = new PDFParser(this, 1);

pdfParser.on("pdfParser_dataError", errData => console.error(errData.parserError));
pdfParser.on("pdfParser_dataReady", pdfData => {
    fs.writeFileSync('./pdf_text.txt', pdfParser.getRawTextContent());
    console.log("Extraction complete");
});

pdfParser.loadPDF("./Catalogo Varejo- Ambtus.pdf");

File: extract_pdf.py
import json
import PyPDF2

pdf_path = "Catalogo Varejo- Ambtus.pdf"

try:
    with open(pdf_path, "rb") as file:
        reader = PyPDF2.PdfReader(file)
        text = ""
        for i in range(len(reader.pages)):
            page_text = reader.pages[i].extract_text()
            text += f"\n--- Page {i+1} ---\n" + page_text
            
        with open("pdf_text.txt", "w", encoding="utf-8") as out:
            out.write(text)
        print("Success")
except Exception as e:
    print(f"Error: {e}")
